HW5/HW5.sql

```diff
@@ -0,0 +1,113 @@
+from pitcher_counts a,
+     pitcher_counts b
+where b.game_id in (select prior_game from rolling_25_prior_days_games where game_id = a.game_id)
+  and a.pitcher = b.pitcher
+group by a.game_id,
+         a.pitcher;
+
+CREATE INDEX pc25 on pitcher_counts_25_day_rolling (game_id);
+
+DROP VIEW IF EXISTS home_team_wins;
+
+CREATE VIEW home_team_wins as
+SELECT g.game_id,
+       year(g.local_date)                                                               year,
+       home.pitcher                                                                     home_starting_pitcher,
+       away.pitcher                                                                     away_starting_pitcher,
+       CASE WHEN b.winner_home_or_away = "H" then 1 else 0 END                       as HOME_TEAM_WINS,                   #response
+       psh.season_avg                                                                   home_starting_pitcher_season_avg, #could use
+       psa.season_avg                                                                   away_starting_pitcher_season_avg, #could use
+       power(psh.season_avg - psa.season_avg, 2)                                        squared_difference_start_pitch_season_avg,
+       #diff in rest days for starting pitcher capped at 7 for no real reason aside from controlling large values
+       case when home.DaysSinceLastPitch > 7 then 7 else home.DaysSinceLastPitch end -
+       case
+           when away.DaysSinceLastPitch > 7 then 7
+           else away.DaysSinceLastPitch end                                          as starting_pitcher_rest_dif_capped,
+       odds.home_line,
+       home_streak.streak - away_streak.streak                                          streak_diff,
+       home_25.hits / home_25.atBats                                                    home_rolling_hits,
+       away_25.hits / away_25.atBats                                                    away_rolling_hits,
+       power(home_25.hits / home_25.atBats - away_25.hits / away_25.atBats, 2)          squared_diff_start_pitch_hits,
+       power(home_25.pitchesThrown - away_25.pitchesThrown, 2)                          squared_diff_start_pitch_thrown,
+       power(home_25.Intent_Walk - away_25.Intent_Walk, 2)                              squared_diff_intent_walk,
+       power(home_25.outsPlayed - away_25.outsPlayed, 2)                                squared_diff_outs_played,
+       CAST(substring(b.temp, 1, length(b.temp) - 8) as int)                            temp,                             #why not
+       power((CAST(substring(home_b.temp, 1, length(home_b.temp) - 8) as int) -
+              CAST(substring(b.temp, 1, length(b.temp) - 8) as int)) -
+             (CAST(substring(away_b.temp, 1, length(away_b.temp) - 8) as int) -
+              CAST(substring(b.temp, 1, length(b.temp) - 8) as int)), 2)
+                                                                                        squared_diff_temp_vs_prior_game
```

Comment:
Might have been more clear to split this query up since this is a lot of from inputs and a lot of when inputs in a row